Add new pop-up modal when image is clicked

PrabSG opened the pull request:
Create a separate modal for displaying the image in a larger size, as well as an accompanying caption, for use on portfolio page's profile image currently. Redesign on header and content to use the more dynamic flexbox layout system.

PrabSG committed:
Add components and styling for profile image modal and add function to
make the modal visible when image is clicked.

Add styling to profile image and appearance of modal's content when it set to appear

Implement general function for setting an image model to open, currently for use on the profile image on portfolio page.

PrabSG committed:
Remove and duplicate text on portfolio page and alter greeting.

PrabSG committed:
Redesign header and content to use flexbox to react more dynamically to page size changes.

DarthGandalf commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,21 @@ function addRandomGreeting() {
+ * Opens a modal with the given image and caption.
+ */
+function openSimpleImageModal(modalId, modalImgId, captionId, closeId, imgSrc, captionText) {  
+  var modal = document.getElementById(modalId);
```
let modal = ...

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/let

DarthGandalf replied:
https://google.github.io/styleguide/jsguide.html#features-local-variable-declarations

PrabSG replied:
Ah yes will change this, is it okay to use const given it is part of ES6?

DarthGandalf replied:
sure

PrabSG replied:
Done

DarthGandalf reviewed (changes requested):
The first 3 commits here are better as a single commit, because they add one feature, and after first commit the webpage doesn't even work properly

DarthGandalf commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,21 @@ function addRandomGreeting() {
+
+  modal.style.display = "block";
+  modalImg.src = imgSrc;
+  caption.innerHTML = captionText;
```
Use `innerText` instead

innerHTML is a very easy way to get XSS

DarthGandalf replied:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/innerText

PrabSG replied:
Will keep this in mind for other sections, thanks. Done

DarthGandalf commented:
please use `git rebase -i master` to fix number of commits (and distribution of code between commits), then `git push -f`

PrabSG commented:
> please use `git rebase -i master` to fix number of commits (and distribution of code between commits), then `git push -f`

Squashed commits into 1 and renamed appropriately, done.
